Sources/EpoxyCollectionView/CollectionView/ReusableViews/CollectionViewCell.swift

```diff
@@ -61,73 +61,108 @@ public final class CollectionViewCell: UICollectionViewCell, ItemCellView {
-    // We correct for this by updating `contentView.bounds`, which updates the constants used by the
-    // width and height constraints created by the `contentView`'s auto-resizing mask.
+      let size: CGSize
+      if horizontalFittingPriority != .required || verticalFittingPriority != .required {
+        // Self-sizing is required in at least one dimension.
+        size = super.systemLayoutSizeFitting(
+          layoutAttributes.size,
+          withHorizontalFittingPriority: horizontalFittingPriority,
+          verticalFittingPriority: verticalFittingPriority)
+      } else {
+        // No self-sizing is required; respect whatever size the layout determined.
+        size = layoutAttributes.size
+      }
 
-    if
-      horizontalFittingPriority == .required &&
-      contentView.bounds.width != layoutAttributes.size.width
-    {
-      contentView.bounds.size.width = layoutAttributes.size.width
+      layoutAttributes.size = size
+      preferredAttributes = layoutAttributes
+    } else {
+      preferredAttributes = super.preferredLayoutAttributesFitting(layoutAttributes)
     }
 
+    // On iOS 15, cells that return an unstable size can result in a layout recursion crash. The
+    // root cause of these crashes is likely an ambiguous layout due to misconfigured constraints.
+    // Unfortunately, finding each and every one of these ambiguous layouts can be challenging. This
+    // code works around this new behavior / crash by giving up on self-sizing after 5 attempts.
+    // Once a component has been sized differently 5 times, we assume that it has an ambiguous
+    // layout and we simply return the last computed size for the remainder of the component's life.
+    // We reset the count on cell reuse and when the cell is reconfigured in `CollectionView`'s
+    // `configure` function.
     if
-      verticalFittingPriority == .required &&
-      contentView.bounds.height != layoutAttributes.size.height
+      CollectionViewConfiguration.shared.enableLayoutRecursionWorkaround,
+      let previousComputedSize = previousComputedSize,
+      numberOfNewComputedSizes >= 5
     {
-      contentView.bounds.size.height = layoutAttributes.size.height
+      preferredAttributes.size = previousComputedSize
     }
 
-    let size: CGSize
-    if horizontalFittingPriority != .required || verticalFittingPriority != .required {
-      // Self-sizing is required in at least one dimension.
-      size = super.systemLayoutSizeFitting(
-        layoutAttributes.size,
-        withHorizontalFittingPriority: horizontalFittingPriority,
-        verticalFittingPriority: verticalFittingPriority)
-    } else {
-      // No self-sizing is required; respect whatever size the layout determined.
-      size = layoutAttributes.size
+    if preferredAttributes.size != previousComputedSize {
+      numberOfNewComputedSizes += 1
+      previousComputedSize = preferredAttributes.size
     }
```

Comment:
Should we guard all of these lines on `CollectionViewConfiguration.shared.enableLayoutRecursionWorkaround`? Seems like we'd be unnecessarily incrementing `numberOfNewComputedSizes` and updating `previousComputedSize` when they'd never be read otherwise